storm-core/src/jvm/org/apache/storm/scheduler/Cluster.java

```diff
@@ -662,22 +662,50 @@ public void setStatusMap(Map<String, String> statusMap) {
         this.status.putAll(statusMap);
     }
 
-    public void setResources(String topologyId, Double[] resources) {
-        this.resources.put(topologyId, resources);
+    /**
+     * Set the amount of resources used used by a topology. Used for displaying resource information on the UI
+     * @param topologyId
+     * @param resources describes the resources requested and assigned to topology in the following format in an array:
+     *  {requestedMemOnHeap, requestedMemOffHeap, requestedCpu, assignedMemOnHeap, assignedMemOffHeap, assignedCpu}
+     */
+    public void setTopologyResources(String topologyId, Double[] resources) {
```

Comment:
Yep, I think that would be helpful. Both for topology and supervisor resources. We may file a separate JIRA for that or do it within this pull request.